pkg/store/cache.go

```diff
@@ -127,8 +127,11 @@ func (c *indexCache) ensureFits(b []byte) bool {
 	if uint64(len(b)) > c.maxSize {
 		return false
 	}
-	for c.curSize+uint64(len(b)) > c.maxSize {
-		c.lru.RemoveOldest()
+	for c.curSize > c.maxSize-uint64(len(b)) {
+		if _, val, ok := c.lru.RemoveOldest(); ok {
```

Comment:
Should we handle something if that does not ok ?

Reply:
How we could inform the users about such case? AFAICT there is no way `ok` can be `false` due to the check before-hand.